### ISSUE TITLE
Links identifiable by colour (hue) must be visually identifiable by some other means.

### ISSUE THREAD

**Issue by mmafe** [ccswbs/hjckrrh#642]
[Old Version of Failure 73](https://www.w3.org/TR/2012/NOTE-WCAG20-TECHS-20120103/F73.html) states "Check that each link within text that is part of a sentence or paragraph (or other area where they could be mistaken as non-linked text) in the page is underlined or otherwise visually identifiable (i.e., bolded, italicized) as a link without relying on color (hue)." under the Test Procedure.

[Current Version of Failure 73](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/F73) states "Check that each link in the page that is identifiable by color (hue) is visually identifiable via some other means (e.g., underlined, bolded, italicized, sufficient difference in lightness, etc)."

This means we would need to review all links in our theme (not just links surrounded by text) and determine if they are identifiable by some other means besides colour (hue).

Issue first identified by Nevan and Brent.

Recommend we review all links on the page, including heading links and determine an appropriate course of action (eg. underlining, changing the colour if we can find a high enough contrast, making things bolder, etc.)

**Comment by nevantan** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [(2) #642 - Links identifiable by colour (hue) must be visually identifiable by some other means.](https://trello.com/c/IUZxj6SI/2184-2-642-links-identifiable-by-colour-hue-must-be-visually-identifiable-by-some-other-means)

**Comment by nevantan** [comment]
Experimenting with colors:
News: #333333; new header color
Features: #666666; current header color

Upcoming Events: #428BCA; new link color (Bootstrap default)
No results found: #2F70A9; current link color

![color_comparisons](https://cloud.githubusercontent.com/assets/25013998/23514124/4fe31a3c-ff34-11e6-9837-df1bade738d3.png)

**Comment by nevantan** [comment]
Final colors found by @morganrees9 are:
Text color: #2C2C2C;
Link color: #017ACD;

**Comment by nevantan** [comment]
## Justification
The color change was made because of WCAG 1.4.1 which states that color must not be the only means of conveying information. Our concern is specifically with [F73](https://www.w3.org/TR/WCAG20-TECHS/F73.html) which is about distinguishable links.

Our current setup is compliant, for the most part, because we use underlines on links to distinguish them from surrounding text. Additionally, all headers on the site - linked or unlinked - were the same color which means no one was able to distinguish them.

Issue #485 made links distinguishable by color and was initially implemented by adding CSS rules for color and underline, the latter to be compliant with WCAG 1.4.1. This solution was technically acceptable from an AODA compliance standpoint, but there was a consensus that underlines on linked headers wouldn't be popular with our clients. To this end, we had to find another way to make linked headers distinguishable from unlinked headers without using underlines.

## Solution
According to [G183](https://www.w3.org/TR/WCAG20-TECHS/G183.html), using a contrast ratio of 3:1 is sufficient for visual distinction. This presented us with a solution that had three criteria:
- Contrast ratio between the link color and text color is at least 3:1 (to meet WCAG 1.4.1 via G183)
- Contrast ratio between the link color and background color is at least 4.5:1 (to meet WCAG 1.4.3)
- New colors should be as similar as possible to current colors to minimize disruption

In the process of searching for a set of colors, @morganrees9 found that the difference between our current text color (#666666) and our current background color (#FFFFFF) didn't allow for a large enough space for there to be a color with sufficient contrast from both. We had to widen the gap by making the text color darker, which allowed us to find the colors listed above.

## Before/After
![link_color_before_after](https://cloud.githubusercontent.com/assets/25013998/23558922/67851922-0003-11e7-97c7-8f956f4e036b.png)

**Comment by mmafe** [comment]
**Update:** Issue found with contrast of the new link colour when set against the footer and breadcrumb background (#F5F5F5). @morganrees9 is investigating.

**Comment by ushenoyuog** [comment]
Closing as pull request is approved. For any changes , please reopen.
